### PR TITLE
[FR] Implement link feature in AppFlowyEditor #886

### PR DESCRIPTION
**Pull request by LucasXu0**
https://github.com/AppFlowy-IO/AppFlowy/issues/886

### CODE REVIEW

File: frontend/app_flowy/packages/appflowy_editor/lib/src/extensions/text_node_extensions.dart [hunk 1]
```diff
@@ -6,22 +6,57 @@ import 'package:appflowy_editor/src/document/text_delta.dart';
...
-      allSatisfyInSelection(StyleKey.strikethrough, true, selection);
+      allSatisfyInSelection(StyleKey.strikethrough, selection, (value) {
+        return value == true;
+      });
 
   bool allSatisfyInSelection(
     String styleKey,
-    dynamic value,
     Selection selection,
+    bool Function(dynamic value) compare,
```
Comment (appflowy) [inline frontend/app_flowy/packages/appflowy_editor/lib/src/extensions/text_node_extensions.dart:59]: Would it be better to use the keyword `predicate`? Just like the filter(predicate: fn ) find(predicate: fn )

File: frontend/app_flowy/packages/appflowy_editor/lib/src/render/link_menu/link_menu.dart [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                ),
+              ]
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildHeader() {
```
Comment (appflowy) [inline frontend/app_flowy/packages/appflowy_editor/lib/src/render/link_menu/link_menu.dart:86]: helper methods or widgets? Check out this [link](https://www.youtube.com/watch?v=IOyq-eTRhvo) .

Comment (appflowy) [reply frontend/app_flowy/packages/appflowy_editor/lib/src/render/link_menu/link_menu.dart:86]: Also, https://inficial.medium.com/flutter-best-practices-for-improve-performance-7e21e14efebb

File: frontend/app_flowy/packages/appflowy_editor/lib/src/render/rich_text/flowy_rich_text.dart [hunk 3]
```diff
@@ -164,43 +189,62 @@ class _FlowyRichTextState extends State<FlowyRichText> with Selectable {
     );
   }
 
-  // unused now.
```
Comment (IMPranshu) [inline frontend/app_flowy/packages/appflowy_editor/lib/src/render/rich_text/flowy_rich_text.dart:167]: Will this be used in the upcoming release? I think keeping this part for future use cases would be an option.

Comment (LucasXu0) [reply frontend/app_flowy/packages/appflowy_editor/lib/src/render/rich_text/flowy_rich_text.dart:167]: No. This part of the code will not be executed in the foreseeable future. 😊

**Pull request merged**